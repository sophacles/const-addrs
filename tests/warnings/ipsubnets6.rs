use const_addrs::ipsubnets6;

fn main() {
    let _ = ipsubnets6!("2001:db8::32:ffff", "2001:db8::32:23", 64);

    // bad one just to fail the compilation and compare output
    let _ = ipsubnets6!("2001:db8::32::ffff", "2001:db8::32:23", 64);
}
