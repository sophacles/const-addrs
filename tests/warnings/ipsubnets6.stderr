warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/ipsubnets6.rs:4:25
  |
4 |     let _ = ipsubnets6!("2001:db8::32:ffff", "2001:db8::32:23", 64);
  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

error: invalid IP address syntax
 --> tests/warnings/ipsubnets6.rs:7:25
  |
7 |     let _ = ipsubnets6!("2001:db8::32::ffff", "2001:db8::32:23", 64);
  |                         ^^^^^^^^^^^^^^^^^^^^
